src/main/java/com/contrastsecurity/maven/plugin/InstallAgentContrastMavenMojo.java

```diff
@@ -8,13 +8,32 @@
 
 import java.io.File;
 import java.text.SimpleDateFormat;
+import java.util.ArrayList;
 import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
 
 
 @Mojo(name = "install", defaultPhase = LifecyclePhase.VALIDATE, requiresOnline = true)
 public class InstallAgentContrastMavenMojo extends AbstractContrastMavenPluginMojo {
     String applicationName;
 
+    static Map<String, String> environmentToSessionMetadata = new TreeMap<String, String>();
+    static {
```

Comment:
Not crazy about a static initializer here. It would prevent us from allowing this to be user configurable.

Reply:
I suppose it could be moved to a property file and loaded from there, but I don't see anyone specifying `-Denvvarname=configurationKey` at Maven runtime to add their own mappings.
At that point it's not much more to create your own session metadata string, as suggested [here](https://github.com/Contrast-Security-OSS/docs/blob/60399b98a20f5729635f7002c4a87983796695cc/content/user/apps/Session-metadata.md#continuous-integration)

Reply:
If someone wanted to have a custom mapping from an environment variable to a session metadata configuration key, they could build their own version of the plugin and that'd be slightly easier to work with if they were in a .properties file I guess?

Reply:
Well they could do it in the maven config in the pom.xml file, the way configuration is done now.